pilot/cmd/istioctl/controlplane/cmd.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// Command returns the "control-plane" subcommand for istioctl.
+func Command(istioNamespaceFlag *string) *cobra.Command {
+	var (
+		features          *[]string
+		out               string
+		helmChartLocation string
+	)
+
+	install := defaultInstall()
+	cmd := &cobra.Command{
+		Use:   "control-plane",
+		Short: "Generates the configuration for Istio's control plane.",
+		Long: "istioctl control-plane produces deployment files to run the minimum Istio control for the set of " +
+			"features requested by the --feature flag. If no features are provided, we create deployments for the " +
+			"default control plane: Pilot, Mixer, CA, an initializer for automatic side-car injection, and Ingress " +
+			"Proxies, with mTLS enabled.",
+		Example: `istioctl control-plane --features routing,policy,initializer -o helm`,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := install.setFeatures(*features); err != nil {
+				return err
+			}
+			install.Namespace = *istioNamespaceFlag
+			switch strings.ToLower(out) {
+			case "helm":
+				_, err := fmt.Fprint(os.Stdout, valuesFromInstallation(install))
+				return err
+			case "yaml":
+				rendered, err := yamlFromInstallation(install, helmChartLocation)
+				if err != nil {
+					return err
+				}
+				_, err = fmt.Fprint(os.Stdout, rendered)
+				return err
+			default:
+				return fmt.Errorf("unsupported output %q", out)
+			}
+		},
+	}
+
+	features = cmd.PersistentFlags().StringArrayP("features", "f", []string{},
+		`List of Istio features to enable. Accepts any combination of "mtls", "telemetry", "routing", "ingress", "policy", "initializer".`)
+	cmd.PersistentFlags().StringVar(&install.Hub, "hub", "gcr.io/istio-testing", "The container registry to pull Istio images from")
+	cmd.PersistentFlags().StringVar(&install.MixerTag, "mixer-tag", "latest", "The tag to use to pull the `mixer` container")
```

Comment:
Production systems should not use 'latest' for image tag.  would it be possible to generate tag based on Istio versions?
 Developers can still set the 'latest' tag and override the default settings.

Reply:
Done.